src/main/java/edu/kis/powp/jobs2d/command/ImmutableCompoundCommand.java

```diff
@@ -39,6 +47,6 @@ public String toString() {
     }
 
     public List<DriverCommand> getCommands() {
-        return commands;
+        return Collections.unmodifiableList(commands);
```

Comment:
returning unmodifiableList may not be enough. Individual commands in the list may be modified. Cloning should solve the problem. Avoid code duplication.